src/tests/test_logger.py

```diff
@@ -37,33 +39,44 @@ class TestLogger(unittest.TestCase):
 
     def test_fileConfig(self):
         """Put logging.dat with special pattern and check it was used"""
-        tmp = os.environ['BITMESSAGE_HOME'] = tempfile.gettempdir()
+        # FIXME: this doesn't always work, probably due to module
+        # load order. Workaround through .travis.yml
+        tmp = set_appdata_folder(tempfile.gettempdir())
         log_config = os.path.join(tmp, 'logging.dat')
         log_file = os.path.join(tmp, 'debug.log')
+        pass_ = 1
 
         def gen_log_config(pattern):
             """A small closure to generate logging.dat with custom pattern"""
             with open(log_config, 'wb') as dst:
                 dst.write(self.conf_template.format(log_file, pattern))
 
-        pattern = r' o_0 '
-        gen_log_config(pattern)
-
         try:
-            from pybitmessage.debug import logger, resetLogging
-            if not os.path.isfile(log_file):  # second pass
-                pattern = r' <===> '
-                gen_log_config(pattern)
-                resetLogging()
+            from pybitmessage.debug import logger, resetLogging, flush_logs
         except ImportError:
             self.fail('There is no package pybitmessage. Things gone wrong.')
-        finally:
-            os.remove(log_config)
+        if not os.path.isfile(log_file):  # second pass
+            pass_ = 2
+            pattern = r' <===> '
+        else:
+            pattern = r' o_0 '
+        gen_log_config(pattern)
+        resetLogging()
 
-        logger_ = logging.getLogger('default')
+        self.longMessage = True
 
-        self.assertEqual(logger, logger_)
+        logger1 = logging.getLogger('default')
 
-        logger_.info('Testing the logger...')
+        self.assertEqual(logger, logger1, ", pass {}".
+                         format(pass_))
 
-        self.assertRegexpMatches(open(log_file).read(), pattern)
+        logger2 = logging.getLogger('root')
+
+        logger2.info('Testing the logger...')
+        flush_logs()
+
+        self.assertRegexpMatches(open(log_file).read(), pattern,
+                                 ", pass {}".format(pass_))
+
+        if pass_ == 2:
+            os.remove(log_config)
```

Comment:
Why only on second pass?

Reply:
I don't understand the logic flow of the passes, this may be a leftover from debugging and perhaps isn't necesasry anymore.

Reply:
This not fixes the test. I'll describe how it works, how it worked and how I wanted it to work a bit later.

Reply:
I'll look at it.

Reply:
> I don't understand the logic flow of the passes, this may be a leftover from debugging and perhaps isn't necesasry anymore.

A proper test should not rely on other tests side effects.

What I wanted to demonstrate in this particular test:
1. when you import `logger` from `debug` it sets up the logging based on `logging.dat` config
1. `logger = logging.getLogger('root')` set after the import will be the same as in `debug`
1. if you call `resetLogging()` the configuration will be reapplied

All that works only after the first import of `debug` because it calls `helper_startup.loadConfig()`.

How it worked before:
1. precondition: there is no other test importing `debug` or other module inducing config setup or appdata change
1. first pass detected by `debug.log` in appdata remaining from core tests
1. on the second pass you need `resetLogging()`

You can reveal it by replacing all `tempfile.gettempdir()` by `tempfile.mkdtemp()` as was suggested before.

Reply:
I'm not sure that I can precisely reconstruct what's happening now. It places `logging.dat` which is used by subsequent runs of pybitmessage process and can be detected on second pass.

Reply:
Maybe subclass this test case from `TestProcessProto` and check only the presence of pattern and maybe severity and initial message? Then we can replace `logging.getLogger('default')` by conventional `logging.getLogger(__name__)`, compose proper `logging.dat` and check presence of all submodule names which write logs.

Reply:
> first pass detected by debug.log in appdata remaining from core tests

Only if it uses the same config directory. But the tests change the location of the config directory. I think this breaks the logic. You can see now that I reverted to your original logic in `test_logger.py`, and it fails on travis-ci.org. I don't want to merge until this is resolved.

Reply:
I wrote before that my original logic was wrong. Standalone tests should not import any module with side effects.
I debug my process-based variant, it succedes only on linux: https://travis-ci.org/github/g1itch/PyBitmessage/builds/740134631